robottelo/ssh.py

```diff
@@ -73,18 +76,28 @@ def _get_connection(
         hostname = settings.server.hostname
     if username is None:
         username = settings.server.ssh_username
-    if key_filename is None:
+    if key_filename is None and password is None:
         key_filename = settings.server.ssh_key
+        # If we don't have key even in settings, read password instead:
+        if key_filename is None:
+            password = settings.server.ssh_password
```

Comment:
I was looking in to **setttings.server** source code, maybe I am missing something, but looks like it is not **lazy-read** the file is parsed and all values loaded by **configure** method which is called once.

So I think the resource saving here is pointless, we do not need to over-complicate using my suggestion above. :)